### PR TITLE
facebook auth moved to social logins lib

### PR DESCRIPTION
**Pull request by zdravkoc**

### CODE REVIEW

File: packages/rx_bloc_cli/mason_templates/bricks/lib_social_logins/__brick__/android/app/src/main/res/values/string.xml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="facebook_app_id">app_id</string>
+    <string name="facebook_client_token">facebook_client_token</string>
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/lib_social_logins/__brick__/android/app/src/main/res/values/string.xml:4]: Please add info section in the readme file that such token needs to be provided

File: packages/rx_bloc_cli/mason_templates/bricks/lib_social_logins/__brick__/android/app/src/main/res/values/string.xml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/lib_social_logins/__brick__/android/app/src/main/res/values/string.xml:2]: I guess the moustache condition needs to be added here

File: packages/rx_bloc_cli/mason_templates/bricks/lib_social_logins/__brick__/lib/lib_social_logins/data_sources/facebook_credential_data_source.dart [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import '../models/facebook_auth_request_model.dart';
+
+class FacebookCredentialDataSource {
+  Future<FacebookAuthRequestModel> getUsersFacebookCredential() async {
+    final result = await FacebookAuth.instance.login(
+      permissions: [
+        'email',
+        'public_profile',
+        'user_birthday',
+        'user_friends',
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/lib_social_logins/__brick__/lib/lib_social_logins/data_sources/facebook_credential_data_source.dart:14]: Please remove the friends file here

File: packages/rx_bloc_cli/mason_templates/bricks/lib_social_logins/__brick__/lib/lib_social_logins/models/facebook_auth_request_model.dart [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import 'package:json_annotation/json_annotation.dart';
+
+part 'facebook_auth_request_model.g.dart';
+
+@JsonSerializable(includeIfNull: false)
+class FacebookAuthRequestModel {
+  final String? email;
+  final String? facebookToken;
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/lib_social_logins/__brick__/lib/lib_social_logins/models/facebook_auth_request_model.dart:8]: Are all fields optional ?

File: packages/rx_bloc_cli/mason_templates/bricks/lib_social_logins/__brick__/lib/lib_social_logins/repositories/facebook_auth_repository.dart [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  final FacebookCredentialDataSource _credentialDataSource;
+
+  Future<AuthTokenModel> facebookAuth(
+          {required FacebookAuthRequestModel requestModel}) =>
+      _errorMapper.execute(
+        () => _fbAuthDataSource.facebookAuthenticate(requestModel),
+      );
+
+  Future<FacebookAuthRequestModel> getUserFacebookCredentials() async {
+    final credentials =
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/lib_social_logins/__brick__/lib/lib_social_logins/repositories/facebook_auth_repository.dart:21]: Is there any reason to not use the error mapper ?

File: packages/rx_bloc_cli/mason_templates/bricks/lib_social_logins/__brick__/lib/lib_social_logins/services/facebook_social_login_service.dart [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  @override
+  Future<AuthTokenModel> authenticate() async {
+    final credential =
+    await _facebookAuthRepository.getUserFacebookCredentials();
+
+    return await _facebookAuthRepository.facebookAuth(requestModel: credential);
+  }
+
+  Future<bool> facebookLogout() async {
+    await FacebookAuth.instance.logOut();
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/lib_social_logins/__brick__/lib/lib_social_logins/services/facebook_social_login_service.dart:28]: Please make the service unit testable!

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/ios/Runner/Info.plist [hunk 7]
```diff
@@ -60,10 +60,22 @@
...
+        <string>fb{app_id}</string>{{/enable_social_logins}}
         <string>$(CF_BUNDLE_URL_SCHEME)</string>
         </array>
         </dict>
-    </array>
+    </array>{{#enable_social_logins}}
+    <key>FacebookAppID</key>
+        <string>{app_id}</string>
+        <key>FacebookClientToken</key>
+        <string>{client_token}</string>
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/ios/Runner/Info.plist:72]: From where those data comes from ?

**Review by StanevPrime — CHANGES_REQUESTED** [review]

File: packages/rx_bloc_cli/mason_templates/bricks/lib_social_logins/__brick__/android/app/src/main/res/values/string.xml (hunk 2 above)
Comment (zdravkoc) [reply packages/rx_bloc_cli/mason_templates/bricks/lib_social_logins/__brick__/android/app/src/main/res/values/string.xml:2]: This file will be generated only if social logins brick is enabled, so do moustache condition really needs to be added here?

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/ios/Runner/Info.plist (hunk 7 above)
Comment (zdravkoc) [reply packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/ios/Runner/Info.plist:72]: User needs to insert app name, app id and client token that he got after he register his app in facebook developer console.
I just added that to readme file with documentation where is all explained

File: packages/rx_bloc_cli/mason_templates/bricks/lib_social_logins/__brick__/android/app/src/main/res/values/string.xml (hunk 2 above)
Comment (StanevPrime) [reply packages/rx_bloc_cli/mason_templates/bricks/lib_social_logins/__brick__/android/app/src/main/res/values/string.xml:2]: Ahh you are right.

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/ios/Runner/Info.plist (hunk 7 above)
Comment (StanevPrime) [reply packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/ios/Runner/Info.plist:72]: I meant something else. The Info.plist reads those variables, which is find. But where they can be set ?

File: packages/rx_bloc_cli/mason_templates/bricks/lib_social_logins/README.md [hunk 8]
```diff
@@ -7,6 +7,12 @@ When you run the created project, don't forget to enable the "Sign in with Apple
 
 Supports iOS.
 
+Facebook authentication uses [flutter_facebook_auth][6] package. In order to make it work you must register your app in facebook developer console, there you will find your <strong>app_id</strong>, <strong>client_token</strong> and <strong>app_name</strong>,
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/lib_social_logins/README.md:10]: Please add a section in the main README.md file as well - > https://pub.dev/packages/rx_bloc_cli/example

File: packages/rx_bloc_cli/mason_templates/bricks/lib_social_logins/__brick__/lib/lib_social_logins/ui_components/facebook_login_widget.dart [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+        ],
+        child: Column(
+          children: [
+            AppErrorModalWidget<SocialLoginBlocType>(
+              errorState: (bloc) => bloc.states.errors,
+            ),
+            RxBlocBuilder<SocialLoginBlocType, bool>(
+                state: (bloc) => bloc.states.isLoading,
+                builder: (context, snapshot, bloc) => snapshot.data == true
+                    ? const SizedBox(
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/lib_social_logins/__brick__/lib/lib_social_logins/ui_components/facebook_login_widget.dart:48]: Please make the loading state consistent across all social login buttons

**Review by StanevPrime — CHANGES_REQUESTED** [review]

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/ios/Runner/Info.plist (hunk 7 above)
Comment (zdravkoc) [reply packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/ios/Runner/Info.plist:72]: User needs to replace this {app_id} with his app id, {client_token} with client token that he gets from facebook console when register new app, its more like placeholders not variables. 
Maybe i should mark them differently?
Or I misunderstood question again?

Comment (StanevPrime) [reply packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/ios/Runner/Info.plist:72]: I'm afraid that you missed only very important concept in the [rx_bloc_cli](https://pub.dev/packages/rx_bloc_cli).

✅ [Build Flavors](https://flutter.dev/docs/deployment/flavors) - Support for multiple flavors (development, staging and production)

From what I see this is not incorporated into the current solution. Let me know if you need information on how to address this comment

File: packages/rx_bloc_cli/example/README.md [hunk 10]
```diff
@@ -24,37 +24,38 @@ Before you start working on your app, make sure you familiarize yourself with th
...
-| `lib/feature_X/blocs` | Feature related [BLoCs][rx_bloc_info_lnk] |
-| `lib/feature_X/di` | Feature related dependencies |
-| `lib/feature_X/services/` | Feature related Services |
-| `lib/feature_X/ui_components/` | Feature related custom widgets |
-| `lib/feature_X/views/` | Feature related pages and forms |
-| `lib/lib_auth/` | The OAuth2 (JWT) based authentication and token management library |
-| `lib/lib_permissions/` | The ACL based library that handles all the in-app routes and custom actions as well. |
-| `lib/lib_router/` | Generally available [router][gorouter_lnk] related classes. The main [router][gorouter_usage_lnk] of the app is `lib/lib_router/routers/router.dart`. |
-| `lib/lib_router/routes` | Declarations of all nested pages in the application are located here |
+| Path                                         | Contains                                                                                                                                              |
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/example/README.md:27]: Is there a reason this section to be changed ?

File: packages/rx_bloc_cli/mason_templates/bricks/lib_social_logins/README.md [hunk 11]
```diff
@@ -7,6 +7,15 @@ When you run the created project, don't forget to enable the "Sign in with Apple
 
 Supports iOS.
 
+Facebook authentication uses [flutter_facebook_auth][6] package. In order to make it work you must register your app in facebook developer console, there you will find your **app_id**, **client_token** and **app_name**,
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/lib_social_logins/README.md:10]: Same as above

File: packages/rx_bloc_cli/mason_templates/bricks/lib_social_logins/__brick__/lib/lib_social_logins/ui_components/facebook_login_widget.dart [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+        child: Column(
+          children: [
+            AppErrorModalWidget<SocialLoginBlocType>(
+              errorState: (bloc) => bloc.states.errors,
+            ),
+            RxBlocBuilder<SocialLoginBlocType, bool>(
+                state: (bloc) => bloc.states.isLoading,
+                builder: (context, snapshot, bloc) => SignInButton(
+                      Buttons.Facebook,
+                      onPressed: () =>
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/lib_social_logins/__brick__/lib/lib_social_logins/ui_components/facebook_login_widget.dart:49]: @JvnSlv let's merge this PR first and you can make the buttons consistent

**Review by StanevPrime — APPROVED** [review]

File: packages/rx_bloc_cli/example/README.md (hunk 10 above)
Comment (zdravkoc) [reply packages/rx_bloc_cli/example/README.md:27]: Section changed when i generated new test project with enabled social logins, we have path `lib/lib_social_logins` but i can revert it to previous version if there is a need

**Review by StanevPrime — APPROVED** [review]

**Pull request merged**